File: src/main/java/com/ultreon/mods/screenshotmanager/common/FloatSize.java
package com.ultreon.mods.screenshotmanager.common;

import lombok.AllArgsConstructor;

import java.util.Objects;

@AllArgsConstructor
public class FloatSize {
    public float width;
    public float height;

    public float getWidth() {
        return width;
    }

    public void setWidth(float width) {
        this.width = width;
    }

    public float getHeight() {
        return height;
    }

    public void setHeight(float height) {
        this.height = height;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        FloatSize size = (FloatSize) o;
        return Float.compare(size.width, width) == 0 && Float.compare(size.height, height) == 0;
    }

    @Override
    public int hashCode() {
        return Objects.hash(width, height);
    }
}

File: common/src/main/java/com/ultreon/mods/screenshotmanager/util/FloatSize.kt
package com.ultreon.mods.screenshotmanager.util

import java.util.*

class FloatSize(var width: Float, var height: Float) {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val size = other as FloatSize
        return size.width.compareTo(width) == 0 && size.height.compareTo(height) == 0
    }

    override fun hashCode(): Int {
        return Objects.hash(width, height)
    }
}